File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3074_apple-redistribution-into-boxes/solutions/java/solution.java
import java.util.Arrays;

class Solution {
    public int minimumBoxes(int[] apple, int[] capacity) {
        Arrays.sort(capacity);
        int totalApples = Arrays.stream(apple).sum();
        int usedBoxes = 0;
        int totalCapacity = 0;

        for (int i = capacity.length - 1; i >= 0; i--) {
            totalCapacity += capacity[i];
            usedBoxes++;
            if (totalCapacity >= totalApples) {
                return usedBoxes;
            }
        }
        
        return usedBoxes; // Ideally, this line should never be needed due to the problem constraints.
    }
}

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3074_apple-redistribution-into-boxes/solutions/cpp/solution.cpp
#include <vector>
#include <algorithm>

class Solution {
public:
    int minimumBoxes(std::vector<int>& apple, std::vector<int>& capacity) {
        int totalApples = 0;
        for (int apples : apple) {
            totalApples += apples;
        }

        std::sort(capacity.rbegin(), capacity.rend());  // Sort in decreasing order
        
        int boxesUsed = 0;
        int currentCapacity = 0;
        for (int cap : capacity) {
            currentCapacity += cap;
            boxesUsed++;
            if (currentCapacity >= totalApples) {
                break;
            }
        }
        
        return boxesUsed;
    }
};
